master/internal/trial.go

```diff
@@ -521,3 +572,54 @@ func mustParseTrialRunID(child *actor.Ref) int {
 	}
 	return id
 }
+
+func (t *trial) maybeRestoreAllocation(ctx *actor.Context) (*model.Allocation, error) {
+	if !t.restored || !config.IsReattachEnabled() {
+		return nil, nil
+	}
+
+	// Do we have an open allocation?
+	var allocations []model.Allocation
+	err := db.Bun().NewSelect().Model(&allocations).
+		Where("task_id = ?", t.taskID).
+		Where("start_time IS NOT NULL").
+		Where("end_time IS NULL").
+		Scan(context.TODO())
+
+	if err != nil {
+		return nil, err
+	}
+
+	openAllocs := len(allocations)
+	switch {
+	case openAllocs == 0:
+		return nil, nil
+	case openAllocs == 1:
+		allocation := &allocations[0]
+		if !config.IsReattachEnabledForRP(allocation.ResourcePool) {
+			return nil, nil
+		}
+
+		return allocation, nil
+	case openAllocs > 1:
+		const maxAllocsToLog int = 3
```

Comment:
`spew.Sdump` is really nice for this kind of stuff

Reply:
that's nice to know. probably the opposite of what I am doing here though, as I am trying to just get a subsample of "bad" results (first three ids) instead of a full detailed dump which ends up being HUGE in the logs.